tests/mqtt_mock_server_handler.h

```diff
@@ -6,6 +6,52 @@
 #include <aws/mqtt/private/packets.h>
 
 static const int MOCK_LOG_SUBJECT = 60000;
+
+struct mqtt_decoded_packet {
+    enum aws_mqtt_packet_type type;
+
+    /* CONNECT */
+    bool clean_session;
+    bool has_will;
+    bool will_retain;
+    bool has_password;
+    bool has_username;
+    uint16_t keep_alive_timeout;
+    enum aws_mqtt_qos will_qos;
+    struct aws_byte_cursor client_identifier;
+    struct aws_byte_cursor will_topic;
+    struct aws_byte_cursor will_message;
+    struct aws_byte_cursor username;
+    struct aws_byte_cursor password;
+
+    /* PUBLISH SUBSCRIBE UNSUBSCRIBE */
+    uint16_t packet_identifier;
+    struct aws_byte_cursor topic_name;         /* PUBLISH topic */
+    struct aws_byte_cursor publish_payload;    /* PUBLISH payload */
+    struct aws_array_list sub_topic_filters;   /* list of aws_mqtt_subscription for SUBSCRIBE */
+    struct aws_array_list unsub_topic_filters; /* list of aws_byte_cursor for UNSUBSCRIBE */
+
+    struct aws_byte_buf buffer;
+};
+
+static int s_mqtt_decode_packet_init(struct mqtt_decoded_packet *packet, struct aws_allocator *alloc) {
```

Comment:
Can we move the implementations to a .c?

Reply:
yep, will do. It's definitely clearer to have a .c file